### PR TITLE
kernel: allow int to enum in only unsafe blocks 

### PR DESCRIPTION
**Pull request by Delta456**
Part of https://github.com/vlang/v/pull/15932

### CODE REVIEW

File: kernel/modules/initramfs/initramfs.v [hunk 1]
```diff
@@ -53,7 +53,9 @@ fn C.string_free(&string)
 
 [cinit]
 __global (
-	volatile module_req = limine.LimineModuleRequest{response: 0}
+	module_req = limine.LimineModuleRequest{
+		response: 0
+	}
```
Comment (spytheman) [inline kernel/modules/initramfs/initramfs.v:58]: why did you change this?

Comment (spytheman) [reply kernel/modules/initramfs/initramfs.v:58]: Is it the result of vfmt? if yes, that is a bug that should be fixed.

Comment (Delta456) [reply kernel/modules/initramfs/initramfs.v:58]: This is from vfmt, yes.

Comment (Delta456) [reply kernel/modules/initramfs/initramfs.v]: Resolved

**Review by spytheman — APPROVED** [review]
